druid-shell/src/keyboard.rs

```diff
@@ -84,6 +84,18 @@ impl KeyEvent {
         }
     }
 
+    /// Map a `KeyEvent`'s `key_code` to by calling f.
+    /// The `KeyEvent` returned leaves all other fields unmodified.
+    pub fn map_keycode(&self, f: fn(KeyCode) -> KeyCode) -> Self {
+        KeyEvent {
+            is_repeat: self.is_repeat,
+            mods: self.mods,
+            key_code: f(self.key_code),
+            text: self.text,
+            unmodified_text: self.unmodified_text,
+        }
+    }
+
```

Comment:
Based on your feedback from to #329,  I tried to include the update event handling stuff into calc,
since this patch is pretty painful to test without it.

doing brought to my attention an ergonomic issue when trying to use this API with `HotKey::matches`.
This was the best thing I could come up with to resolve that, however leaving `text` and `unmodified_text` untouched seems pretty terrible.